escrow/test/Escrow.js

```diff
@@ -0,0 +1,106 @@
+const { expect } = require("chai");
+const { Contract, ContractFactory } = require("ethers");
+const { calcEthereumTransactionParams } = require("@acala-network/eth-providers");
+const { ACA } = require("@acala-network/contracts/utils/Address");
+
+const TokenContract = require("@acala-network/contracts/build/contracts/Token.json");
+
+const txFeePerGas = '199999946752';
+const storageByteDeposit = '100000000000000';
+
+const EscrowContract = require("../artifacts/contracts/escrow.sol/Escrow.json");
+
+describe("Escrow contract", function () {
+  const ethParams = calcEthereumTransactionParams({
+    gasLimit: '21000010',
+    validUntil: '360001',
+    storageLimit: '64001',
+    txFeePerGas,
+    storageByteDeposit
+  });
+
+  let deployer;
+  let requestor;
+  let provider;
+  let requestorAddress;
+  let providerAddress;
+  let acaInstance;
+  let Escrow;
+  let escrowInstance;
+
+  const ESCROW_AMOUNT = 1_000_000_000; // 0.001 ACA
+
+  beforeEach(async function () {
+    [deployer, requestor, provider] = await ethers.getSigners();
+
+    requestorAddress = await requestor.getAddress();
+    providerAddress = await provider.getAddress();
+
+    acaInstance = new Contract(ACA, TokenContract.abi, requestor);
+
+    Escrow = new ContractFactory(EscrowContract.abi, EscrowContract.bytecode, deployer);
+    escrowInstance = await Escrow.deploy(acaInstance.address, ESCROW_AMOUNT, requestorAddress, providerAddress,
+      {
+        gasPrice: ethParams.txGasPrice,
+        gasLimit: ethParams.txGasLimit,
+      }
+    );
+  });
+
+  it("deployment should assign the escrow provider and requestor", async function () {
+    expect(await escrowInstance.requestor()).to.equal(requestorAddress);
+    expect(await escrowInstance.serviceProvider()).to.equal(providerAddress);
+  });
```

Comment:
Should we check for tokenAddress and amount too?

It's indirectly tested later on tho

Reply:
This might be beneficial, yes. It seems as one of the important direct checks. In tutorials we are lenient with the test suite, as in we don't need to cover everything, but the important things should still be validated.